python/tank/pipelineconfig_utils.py

```diff
@@ -32,6 +32,46 @@
 logger = LogManager.get_logger(__name__)
 
 
+class ConfigFileResolver(object):
```

Comment:
This logic needs to be taken out of the pipeline configuration because we'll probably need to reused it in other contexts where we don't have a pipeline configuration object...